tests/test_router_bgp.rb

```diff
@@ -341,37 +341,33 @@ def test_event_history
     opts = [:cli, :detail, :events, :periodic]
     opts.each do |opt|
       # Test basic true
-      bgp.send("event_history_#{opt}_set", true)
+      bgp.send("event_history_#{opt}=", 'true')
+      set = bgp.send("default_event_history_#{opt}")
       result = bgp.send("event_history_#{opt}")
-      assert(result, 'Failed to set state to True')
+      assert_equal(set, result,
+                   'Failed to set True with Size')
 
       # Test true with size
-      bgp.send("event_history_#{opt}_set", true, 'large')
-      result = bgp.send("event_history_#{opt}_size")
-      assert_equal('large', result,
+      bgp.send("event_history_#{opt}=", 'size_large')
+      result = bgp.send("event_history_#{opt}")
+      assert_equal('size_large', result,
                    'Failed to set True with Size')
 
       # Test false with size
-      bgp.send("event_history_#{opt}_set", false)
+      bgp.send("event_history_#{opt}=", 'false')
```

Comment:
It is designed as a string property. I do not think it can support both. If I set no event_history cli with false or 'false', when get it, it does not know whether to return false or 'false'. Then it does not match in the puppet.